### PR TITLE
[Swift APIView] Create intermediate models and SwiftAPIViewTests package

### LINKED ISSUE #3091: [Swift APIView] Improve TypeModel implementation
Currently this is a huge, ugly, messy chunk of code. A cleaner implementation would be to have separate structs for each kind of supported Type annotation. From the Swift grammar docs:
- [x] function-type
- [x] array-type
- [x] dictionary-type
- [x] type-identifier
- [x] tuple-type
- [x] optional-type
- [x] implicitly-unwrapped-optional-type
- [x] protocol-composition-type
- [x] opaque-type
- [x] metatype-type
- [x] any-type
- [x] self-type

At least two types display with the current type model in a buggy way:
- tuple-identifier
- protocol-composition-type

### PR DESCRIPTION
**Pull request by tjprescott**
This refactor extracts tests out to a `SwiftAPIViewTests` package a la Python, adds a number of new tests from Apple's own documentation, and adds an internal intermediate model that will allow us to manipulate objects in support of fixing the issues we see with extensions. Fixes #3091.

This PR does _not_ attempt to fix the issue with extensions. It just lays that groundwork. 

APIView: https://apiviewstaging.azurewebsites.net/Assemblies/Review/a0490ca69cf740cea3ce6fabad9c3c67/2b0384fd32894ecfa24edfcd73a11cff

### CODE REVIEW

File: src/swift/SwiftAPIViewCore/Sources/Models/APIViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+    func add(token: NavigationToken) {
+        self.navigation.append(token)
+    }
+
+    func text(_ text: String, definitionId: String? = nil) {
+        checkIndent()
+        let item = Token(definitionId: definitionId, navigateToId: nil, value: text, kind: .text)
+        // TODO: Add cross-language definition ID
+        // if add_cross_language_id:
```
Comment (mpodwysocki) [inline src/swift/SwiftAPIViewCore/Sources/Models/APIViewModel.swift:131]: Is this just for Swift or across languages?

File: src/swift/SwiftAPIViewCore/Sources/Models/Declarations/ExtensionModel.swift [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        attributes = AttributesModel(from: decl.attributes)
+        accessLevel = decl.accessLevel
+        self.typeModel = decl.type.toTokenizable()!
+        typeInheritanceClause = TypeInheritanceModel(from: decl.typeInheritanceClause)
+        genericWhereClause = GenericWhereModel(from: decl.genericWhereClause)
+        members = [AccessLevelProtocol]()
+        decl.members.forEach { member in
+            switch member {
+            case let .declaration(decl):
+                // TODO: We need to push these into the relevant object definitions.
```
Comment (mpodwysocki) [inline src/swift/SwiftAPIViewCore/Sources/Models/Declarations/ExtensionModel.swift:52]: What does this mean? Push the extension declaration into the proper Extended class?

**Review by mpodwysocki — APPROVED** [review]
LGTM

File: src/swift/SwiftAPIViewCore/Sources/Models/APIViewModel.swift (hunk 1 above)
Comment (tjprescott) [reply src/swift/SwiftAPIViewCore/Sources/Models/APIViewModel.swift:131]: I just left comment here so I know where to add it. Right now only Python has this implemented.

File: src/swift/SwiftAPIViewCore/Sources/Models/APIViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+        add(token: item)
+    }
+
+    func member(name: String, definitionId: String? = nil) {
+        let item = Token(definitionId: definitionId, navigateToId: nil, value: name, kind: .memberName)
+        add(token: item)
+    }
+
+    // TODO: Add support for diagnostics
+//    func diagnostic(self, text, line_id):
```
[the remark this answers is not in the document]
Comment (tjprescott) [reply src/swift/SwiftAPIViewCore/Sources/Models/APIViewModel.swift:259]: They've never been implemented in Swift APIView, so I added this to come to later.

File: src/swift/SwiftAPIViewCore/Sources/Models/Declarations/ExtensionModel.swift (hunk 2 above)
Comment (tjprescott) [reply src/swift/SwiftAPIViewCore/Sources/Models/Declarations/ExtensionModel.swift:52]: It was more me thinking out loud. Part of: https://github.com/Azure/azure-sdk-tools/issues/2536


I'm not yet sure what the best way would be.

**Pull request merged**